JS beautify break the angular compile

# Description
I use the js-beautify to beauty my angular 6 code generated by angular-cli. However, for a file called 'test.ts', js-beautify will break a statement: 
```
declare const require: any;
```

into two lines:

 ```
declare
const require: any;
```

which will break the angular compile:

 > ERROR in src/test.ts(12,1): error TS2304: Cannot find name 'declare'.
 > src/test.ts(13,7): error TS1155: 'const' declarations must be initialized.

# Steps to Reproduce
1. use angular-cli to create a new project
 > npm install angular-cli
 > ng new demo-project
2. then use the js-beautify to beauty the file demo-project/src/test.ts
compile the project
 > npm install
 > npm start

Then we will get the error message.

## Environment
OS: Debian 9, Node 8.11.3 npm 6.3.0, angular 6.1.3
## Settings
no settings, all default
run like 

 > js-beautify -r [outputfile]

--- a/js/src/javascript/beautifier.js
+++ b/js/src/javascript/beautifier.js
@@ -866,6 +866,9 @@ Beautifier.prototype.handle_word = function(current_token) {
         this._output.space_before_token = true;
       } else if (reserved_word(this._flags.last_token, 'default') && this._last_last_text === 'export') {
         this._output.space_before_token = true;
+      } else if (this._flags.last_token.text === 'declare') {
+        // accomodates Typescript declare function formatting
+        this._output.space_before_token = true;
       } else {
         this.print_newline();
       }
@@ -953,6 +956,9 @@ Beautifier.prototype.handle_word = function(current_token) {
   } else if (prefix === 'NEWLINE') {
     if (reserved_array(this._flags.last_token, special_words)) {
       // no newline between 'return nnn'
+      this._output.space_before_token = true;
+    } else if (this._flags.last_token.text === 'declare' && reserved_array(current_token, ['var', 'let', 'const'])) {
+      // accomodates Typescript declare formatting
       this._output.space_before_token = true;
     } else if (this._flags.last_token.type !== TOKEN.END_EXPR) {
       if ((this._flags.last_token.type !== TOKEN.START_EXPR || !reserved_array(current_token, ['var', 'let', 'const'])) && this._flags.last_token.text !== ':') {

--- a/js/test/generated/beautify-javascript-tests.js
+++ b/js/test/generated/beautify-javascript-tests.js
@@ -3748,6 +3748,17 @@ function run_javascript_tests(test_obj, Urlencoded, js_beautify, html_beautify, 
             '        return 0;\n' +
             '    }\n' +
             '}');
+        
+        // Issue 1544 - Typescript declare formatting (no newline).
+        bt(
+            'declare const require: any;\n' +
+            'declare function greet(greeting: string): void;\n' +
+            'declare var foo: number;\n' +
+            'declare namespace myLib {\n' +
+            '    function makeGreeting(s: string): string;\n' +
+            '    let numberOfGreetings: number;\n' +
+            '}\n' +
+            'declare let test: any;');
         bt(
             'interface Test {\n' +
             '    blah: string[];\n' +

--- a/python/jsbeautifier/javascript/beautifier.py
+++ b/python/jsbeautifier/javascript/beautifier.py
@@ -782,6 +782,9 @@ class Beautifier:
                     self._output.space_before_token = True
                 elif reserved_word(self._flags.last_token, 'default') and self._last_last_text == 'export':
                     self._output.space_before_token = True
+                elif self._flags.last_token.text == 'declare':
+                    # accomodates Typescript declare function formatting
+                    self._output.space_before_token = True
                 else:
                     self.print_newline()
             elif self._flags.last_token.type == TOKEN.OPERATOR or self._flags.last_token.text == '=':
@@ -857,6 +860,12 @@ class Beautifier:
         elif prefix == 'NEWLINE':
             if reserved_array(self._flags.last_token, _special_word_set):
                 # no newline between return nnn
+                self._output.space_before_token = True
+            elif self._flags.last_token.text == 'declare' and reserved_array(current_token, [
+                    'var',
+                    'let',
+                    'const']):
+                # accomodates Typescript declare formatting
                 self._output.space_before_token = True
             elif self._flags.last_token.type != TOKEN.END_EXPR:
                 if (

--- a/python/jsbeautifier/tests/generated/tests.py
+++ b/python/jsbeautifier/tests/generated/tests.py
@@ -3504,6 +3504,17 @@ class TestJSBeautifier(unittest.TestCase):
             '        return 0;\n' +
             '    }\n' +
             '}')
+        
+        # Issue 1544 - Typescript declare formatting (no newline).
+        bt(
+            'declare const require: any;\n' +
+            'declare function greet(greeting: string): void;\n' +
+            'declare var foo: number;\n' +
+            'declare namespace myLib {\n' +
+            '    function makeGreeting(s: string): string;\n' +
+            '    let numberOfGreetings: number;\n' +
+            '}\n' +
+            'declare let test: any;')
         bt(
             'interface Test {\n' +
             '    blah: string[];\n' +

--- a/test/data/javascript/tests.js
+++ b/test/data/javascript/tests.js
@@ -2288,6 +2288,19 @@ exports.test_data = {
           ]
         },
         {
+          comment: "Issue 1544 - Typescript declare formatting (no newline).",
+          unchanged: [
+            'declare const require: any;',
+            'declare function greet(greeting: string): void;',
+            'declare var foo: number;',
+            'declare namespace myLib {',
+            '    function makeGreeting(s: string): string;',
+            '    let numberOfGreetings: number;',
+            '}',
+            'declare let test: any;'
+          ]
+        },
+        {
           unchanged: [
             'interface Test {',
             '    blah: string[];',

